backend/substrapp/ledger_utils.py

```diff
@@ -157,6 +161,50 @@ def _wrapper(*args, **kwargs):
     return _retry
 
 
+def to_snake_case(camel_str):
+    name = CAMEL_TO_SNAKE_PATTERN.sub(r'\1_\2', camel_str)
+    name = CAMEL_TO_SNAKE_PATTERN_2.sub(r'\1_\2', name).lower()
+    return name.replace('_i_d', '_id')
```

Comment:
I think something like 
```python 
def to_snake_case(camel_str):
    if len(camel_str) <= 2:
        return camel_str.lower()
    
    name = CAMEL_TO_SNAKE_PATTERN.sub(r'\1_\2', camel_str)
    name = CAMEL_TO_SNAKE_PATTERN_2.sub(r'\1_\2', name).lower()
    return name
```
Will englobe more cases and not only 'iD', what do you think?

Reply:
Nice idea ! But as we will do the update (camel to snake case) in the chaincode, this function will be removed in the future.
I don't know if it is worth to do this update in that context !

Reply:
And this case is to handle `IDToKey` and `ComputePlanID`, so you suggestion won't work in those cases no ?

Reply:
It looks that it works, but anyway if this is a temporary code, fine for me also to keep it like this
```python
import re

CAMEL_TO_SNAKE_PATTERN = re.compile(r'(.)([A-Z][a-z]+)')
CAMEL_TO_SNAKE_PATTERN_2 = re.compile(r'([a-z0-9])([A-Z])')

def to_snake_case(camel_str):
    if len(camel_str) <=2:
        return camel_str.lower()
    
    name = CAMEL_TO_SNAKE_PATTERN.sub(r'\1_\2', camel_str)
    name = CAMEL_TO_SNAKE_PATTERN_2.sub(r'\1_\2', name).lower()
    return name


print(to_snake_case('iD'))
print(to_snake_case('IDToKey'))
print(to_snake_case('ComputePlanID'))
```

Reply:
Yes, I found this was made for `'authorizedIDs'`

Reply:
I understand now :-). Let forget about my comment, and remove this after the chaincode update

Reply:
Removal incoming: https://github.com/SubstraFoundation/substra-chaincode/pull/125